client/components/chart/index.js

```diff
@@ -0,0 +1,213 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import { Component, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import D3Chart from './charts';
+import dummyOrders from './test/fixtures/dummy';
+import Legend from './legend';
+
+class WidgetCharts extends Component {
+	constructor() {
+		super( ...arguments );
+		this.getOrderedKeys = this.getOrderedKeys.bind( this );
+		this.handleLegendHover = this.handleLegendHover.bind( this );
+		this.handleLegendToggle = this.handleLegendToggle.bind( this );
+		this.updateDimensions = this.updateDimensions.bind( this );
+		this.handleChange = this.handleChange.bind( this );
+		this.getSomeOrders = this.getSomeOrders.bind( this );
+		const products = [
+			{
+				key: 'date',
+				selected: true,
+			},
+			{
+				key: 'Cap',
+				selected: true,
+			},
+			{
+				key: 'T-Shirt',
+				selected: true,
+			},
+			{
+				key: 'Sunglasses',
+				selected: true,
+			},
+			{
+				key: 'Polo',
+				selected: true,
+			},
+			{
+				key: 'Hoodie',
+				selected: true,
+			},
+		];
+		const someOrders = this.getSomeOrders( products );
+		this.state = {
+			products,
+			orderedKeys: this.getOrderedKeys( someOrders ).map( d => ( {
+				...d,
+				visible: true,
+				focus: true,
+			} ) ),
+			someOrders,
+			bodyWidth: document.getElementById( 'wpbody' ).getBoundingClientRect().width,
+		};
+	}
+
+	getSomeOrders( products ) {
+		return dummyOrders.map( d => {
+			return Object.keys( d )
+				.filter( key =>
+					products
+						.filter( k => k.selected )
+						.map( k => k.key )
+						.includes( key )
+				)
+				.reduce( ( accum, current ) => ( { ...accum, [ current ]: d[ current ] } ), {} );
+		} );
```

Comment:
When we hook this up to an api, it will make sense to persist this kind of operation somehow.